### PR TITLE
Homepage & general application styling - prettier additions

### PR DESCRIPTION
**Pull request by michaelgee22**

### CODE REVIEW

File: src/css/homepage.css
Comment (BenDMyers) [inline src/css/homepage.css:38]: I ran a [color contrast check](https://webaim.org/resources/contrastchecker/) between this color and the white text, and I'm getting a contrast ratio of **3.32:1**. [Text this size should meet a ratio of **4.5:1**](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).

We can remedy our contrast two ways:

* We could find a darker, more contrasting background color.
* We could make the link contents "large text," which only needs to meet a **3:1** ratio. Large text can be either:
    * At least `18pt`, or
    * At least `14pt` and bold

I'm happy to leave it up to you to decide how you'd like to go about handling this contrast.

File: src/css/homepage.css
Comment (BenDMyers) [inline src/css/homepage.css:42]: I ran a color contrast check here too, and the white-on-gray has a color contrast ratio of **2.87:1**. This means that even if our text is larger, it would still not meet the required **3:1** color contrast ratio for large text. We should darken this gray somewhat, though how much darker it needs to be depends on whether you want to make the link text bigger.

Like with the upcoming event color contrast, I'm fine no matter how you want to handle this.

File: src/_includes/layouts/homepage.html
Comment (BenDMyers) [inline src/_includes/layouts/homepage.html:24]: I love the markup here! You've done a great job! I wanted to propose a bit of [ARIA](https://benmyers.dev/blog/aria) that might help make this even better for screenreader users. Specifically, if we use ARIA to _label_ these articles, it'll make it much easier for screenreader users to hop from `article` to `article`.

[We can use the `h3`s as our label](https://benmyers.dev/blog/aria-labels-and-descriptions/#aria-labelledby) — here's the flow I'm thinking:

1. Give the `h3` an `id` - we could use [Eleventy's built-in `slug` filter](https://www.11ty.dev/docs/filters/slug/) to generate it from the `title` frontmatter.
2. Then, we set the `article`'s `aria-labelledby` attribute to the `h3`'s `id`.

File: src/_includes/layouts/homepage.html
Comment (BenDMyers) [inline src/_includes/layouts/homepage.html:37]: We can do the `aria-labelledby` stuff here, too.

File: src/_includes/layouts/homepage.html
Comment (BenDMyers) [inline src/_includes/layouts/homepage.html:29]: Some screenreader users navigate by skipping from link to link. In that case, though, they'd hear "View Details, link. View Details, link. View Details, link." and so forth on our page. While sighted users know those links are different, because the links are close to the event titles, blind users who use screenreaders don't necessarily get that context in all navigation modes. Because of that, [it's best to make the link announcement distinct for each link.](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)

I recommend using `aria-label` here, so that we can curate the screenreader experience without necessarily changing the visual experience. Something like:

```html
<a href="{{ event.url }}" aria-label="View Details for {{ event.data.title }}">View Details &#8250;</a>
```

What do you think?

**Comment by michaelgee22** [comment]
wow, a ton of great feedback that quick, I'll get right on it!

I've never used aria-labelledby so this is a good learning point for me. 

also, I see dates for each item are added in the other PR so I can address that as well when that goes in or later whatever you prefer

File: src/_includes/layouts/homepage.html
Comment (michaelgee22) [inline src/_includes/layouts/homepage.html:42]: @BenDMyers I wrapped the event title in single quotes in the aria-label here, what do you think? unnecessary?

**Comment by michaelgee22** [comment]
learned a lot from this PR, let me know if you have any other feedback!

File: src/_includes/layouts/homepage.html
Comment (BenDMyers) [reply src/_includes/layouts/homepage.html:42]: Probably unnecessary, I think. Depending on the screenreader user's verbosity settings, it might even be a bit of extra clutter, since verbose screenreaders might announce something like "View Details for apostrophe Eleventy at 11 apostrophe, link." I'd suggest removing the apostrophes.

**Comment by chantastic** [comment]
WOW! What collaboration.
This was wicked fun to catch up on! I love seeing both the updated layout AND the accessibility discussion.
Thank you @michaelgee22 and @BenDMyers 💞

**Review by chantastic — APPROVED** [review]
I love it!
If there are no objections, let's merge this before the weekend and we can add any iterations in new PRs :)

**Review by BenDMyers — APPROVED** [review]
I'm happy with this! But I'll let anyone else who wants to comment weigh in.

File: .prettierrc
Comment (msutkowski) [inline .prettierrc:5]: Obligatory spaces vs tabs comment.

**Review by msutkowski — APPROVED** [review]
Looks good to me. Awesome turnaround time!

**Comment by michaelgee22** [comment]
sounds good! I'm out at dinner right now but will merge first thing once I get home

File: .prettierrc
Comment (michaelgee22) [reply .prettierrc:5]: I'm for sure team tabs but when I set it to true it indented the code all the way to Europe. Maybe it should be true with tabWidth "1"?

Comment (BenDMyers) [reply .prettierrc:5]: I'm typically Team Tabs as well, particularly for publicly shared code and namely for accessibility reasons. The custom tabwidth settings in editors are easier for people who use screen magnifiers (who might want to set a small tabwidth so their own tabs don't indent all the way to Europe). Additionally, each tab and/or space takes up a character in a braille reader, which usually has limited room, so one tab tends to be more economical for braille reader users than 2, 4, 8, whatever number of spaces.

It's a pity that GitHub's tabwidth is 8, though.

Comment (michaelgee22) [reply .prettierrc:5]: makes sense. Will fix this before merging!

Comment (chantastic) [reply .prettierrc:5]: i'd be interested in working in a tabs project.
i've always liked tabs philosophically but have alway worked in Ruby projects where `two spaces` like a god

**Comment by BenDMyers** [comment]
@all-contributors please add @michaelgee22 for accessibility, code, and design

**Pull request merged**
